swarm/src/lib.rs

```diff
@@ -545,7 +548,9 @@ where
 
     /// Returns an iterator that produces the list of addresses we're listening on.
     pub fn listeners(&self) -> impl Iterator<Item = &Multiaddr> {
-        self.listeners.listen_addrs()
+        // TODO
+        vec![].into_iter()
+        // self.transport.listen_addrs()
     }
```

Comment:
Should the Transport track the listening addresses or should the swarm do this according to new / expired addresses reported in `TransportEvent`s?

Reply:
I think that is a `Swarm` responsibility.